From 51faf40f94b894ae181bad80bae402a690843623 Mon Sep 17 00:00:00 2001
From: Wang Zaikuo <zaikuo.wang@intel.com>
Date: Tue, 31 Jul 2018 14:58:26 +0800
Subject: [PATCH 24/54] media: intel-ipu4: crlmodule: set routing 0 immutable

Message for Open Source:
Set routing 0 immutable and always active because
there must be one default routing for pipeline and
it can not be disabled.

Message for Internal:
[Issue/Feature] Same as above
[Root Cause/Changes] Same as above

Change-Id: I4d78bcf6bb765c07a749de2aa47b486bcbe95515
Tracked-On: #H1506250490
Signed-off-by: Wang Zaikuo <zaikuo.wang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/crlmodule/crlmodule-core.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/crlmodule/crlmodule-core.c b/drivers/media/i2c/crlmodule/crlmodule-core.c
index a4cd4bc..efea035 100755
--- a/drivers/media/i2c/crlmodule/crlmodule-core.c
+++ b/drivers/media/i2c/crlmodule/crlmodule-core.c
@@ -2443,6 +2443,10 @@ static int crlmodule_set_routing(struct v4l2_subdev *subdev,
 		    t->sink_pad != CRL_PAD_SINK)
 			continue;
 
+		if (sensor->src->route_flags[t->source_stream] &
+			V4L2_SUBDEV_ROUTE_FL_IMMUTABLE)
+			continue;
+
 		if (t->flags & V4L2_SUBDEV_ROUTE_FL_ACTIVE)
 			sensor->src->route_flags[t->source_stream] |=
 				V4L2_SUBDEV_ROUTE_FL_ACTIVE;
@@ -2849,7 +2853,8 @@ static int crlmodule_init_subdevs(struct v4l2_subdev *subdev)
 				sd->route_flags[j] =
 					V4L2_SUBDEV_ROUTE_FL_SOURCE;
 			sd->route_flags[0] |=
-					V4L2_SUBDEV_ROUTE_FL_ACTIVE;
+					V4L2_SUBDEV_ROUTE_FL_ACTIVE |
+					V4L2_SUBDEV_ROUTE_FL_IMMUTABLE;
 		}
 
 		sd->sd.entity.ops = &crlmodule_entity_ops;
-- 
1.9.1

